### DISCUSSION TITLE
Built-in Eslint + Prettier

### DISCUSSION THREAD

**Discussion by adriandmitroca** [vercel/next.js#14002] [Ideas]
Any actual reasons why Eslint and Prettier isn't included out of the box with Next?

I suppose that's something mandatory nowadays and definitely majority of the projects use it anyways, which leads to configuring it again over and over in every single project.

Sensible defaults would be great, especially when you consider that they could include some Next.js specific rules that would simply help development.

**Comment by jamesmosier** [comment]
I believe an intergration with ESLint might be in the works. 

As an aside, there is also a Next specific [ESLint](https://github.com/vercel/next.js/tree/canary/packages/eslint-plugin-next) plugin being worked on.

**Comment by bogdansoare** [comment]
Having built-in eslint + prettier would be great

**Comment by mohsen1** [comment]
Think about new comers. It is definitely a hurdle for new comers to deal with lint errors.

**Reply by adriandmitroca** [reply]
I hardly disagree. This can be enabled as opt-in, also, most likely it fixes more problems than it might create.

**Reply by mohsen1** [reply]
I am +1 for opt in. Though with the work on plugins this seems like a good plugin candidate then a core feature.

**Comment by FDiskas** [comment]
I'm usually doing like so:
```
npx mrm prettier
npx mrm eslint
npm i -D eslint-plugin-prettier
```
And add some more settings in `.eslintrc.json`

```
    {
      "plugins": ["prettier"],
      "rules": {
        "prettier/prettier": "error",
      }
    }
```

And I don't agree that this should be integrated. Many companies has their own rules and strategies for code quality.
